src/vcpkg/commands.export-port.cpp

```diff
@@ -0,0 +1,280 @@
+#include <vcpkg/base/git.h>
+#include <vcpkg/base/hash.h>
+#include <vcpkg/base/json.h>
+#include <vcpkg/base/system.debug.h>
+#include <vcpkg/base/util.h>
+
+#include <vcpkg/archives.h>
+#include <vcpkg/commands.export-port.h>
+#include <vcpkg/configuration.h>
+#include <vcpkg/input.h>
+#include <vcpkg/metrics.h>
+#include <vcpkg/paragraphs.h>
+#include <vcpkg/registries.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace
+{
+    using namespace vcpkg;
+
+    void print_success_message(Path destination,
+                               StringView port_name,
+                               Optional<Version> version,
+                               Optional<StringView> registry = nullopt)
+    {
+        if (!version)
+        {
+            msg::println(msgExportPortSuccessUnversioned, msg::path = destination, msg::package_name = port_name);
+        }
+        else if (auto r = registry.get())
+        {
+            msg::println(msgExportPortSuccessFromRegistry,
+                         msg::path = destination,
+                         msg::package_name = port_name,
+                         msg::version = version.value_or_exit(VCPKG_LINE_INFO),
+                         msg::url = *r);
+        }
+        else
+        {
+            msg::println(msgExportPortSuccess,
+                         msg::path = destination,
+                         msg::package_name = port_name,
+                         msg::version = version.value_or_exit(VCPKG_LINE_INFO));
+        }
+    }
+
+    Optional<Version> try_get_port_version(Filesystem& fs, const Path& destination)
+    {
+        auto maybe_spgh = Paragraphs::try_load_port(fs, destination);
+        if (auto spgh = maybe_spgh.get())
+        {
+            if (auto scf = spgh->get())
+            {
+                return scf->to_version();
+            }
+        }
+        return nullopt;
+    }
+
+    void copy_port_files(Filesystem& fs, StringView port_name, const Path& source, const Path& destination)
+    {
+        std::error_code ec;
+        if (!fs.exists(source, VCPKG_LINE_INFO))
+        {
+            Checks::msg_exit_with_error(
+                VCPKG_LINE_INFO, msgExportPortFilesMissing, msg::package_name = port_name, msg::path = source);
+        }
+
+        auto port_files = fs.get_regular_files_recursive(source, ec);
+        if (ec)
+        {
+            Checks::msg_exit_with_error(
+                VCPKG_LINE_INFO, msgExportPortFilesMissing, msg::package_name = port_name, msg::path = source);
+        }
+
+        for (const auto& file : Util::fmap(port_files, [&source](StringView&& str) -> Path {
+                 return str.substr(source.generic_u8string().size() + 1);
```

Comment:
1. `generic_u8string` is loop-invariant so it should be sucked out of the loop.
2. I'm not sure what converting `\`s into `/`s is doing for this; why not `native`?
3. It seems like it would be better to just do this as the first thing in the loop, such as:
    ```
    for (const auto& port_file : port_files) {
        const auto file = port_file.substr(source.generic_u8string().size() + 1);
    ```